### DISCUSSION TITLE
Unable to use the `spheres_and_cylinders` size factor model

### DISCUSSION THREAD

**Discussion by JahanzebTariq** [PMEAL/OpenPNM#2651] [Q&A]
Greeting, @jgostick  and @ma-sadeghi  I am trying to use the module named **Spheres and Cylinder  for the hydraulic size factor** but not being able to do it because the **throat diameter is greater than the pore diameter**. Please address this issue, so that I may proceed further

`pn.add_model(propname='throat.hydraulic_size_factors',model=op.models.geometry.hydraulic_size_factors.cones_and_cylinders)`

In this code I am given the restriction of **not being able to use Spheres and Cylinders instead of Cones and Cylinders**. Kindly address my issue and **please do tell me the reason**. Thank you

**Comment by ma-sadeghi** [comment]
As the error message says: when throat diameter is greater than the diameter of adjacent pores, you can't "physically" fit a stick and ball conduit in that location. 

In previous versions, we used to catch and "fix" these unphysical circumstances by applying some opinionated rule, but in version 3, we decided to not do that anymore.

**Reply by JahanzebTariq** [reply]
so sir **what would you recommend**.. should I go with **cones_and_cylinders**.. and if so **why is it**..? meaning is it a **trail and error method to get a good result**.. because **previous code** used **pyramids and cones** in this place.. is it **users discretion**  to use what ever he likes..? @ma-sadeghi

**Reply by ma-sadeghi** [reply]
As far as I know, choosing the right size factor model is a trial and error process. I've always wanted to do a thorough benchmark of all available size factor models for both flow and diffusion, but I don't see it happening in the forseeable future.

**Reply by JahanzebTariq** [reply]
So According to you if **I am asked about the selection procedure, then I would answer the above statement**..? please sir @ma-sadeghi

**Reply by ma-sadeghi** [reply]
That's how I would respond. Of course, if your actual geometry "looks" like a stick and ball model, then in principle, `spheres_and_cylinders` should give the best result.

**Reply by JahanzebTariq** [reply]
@ma-sadeghi .. Thank you sir.. **You people are really Awesome** .. I can't thank you enough..
